### ISSUE TITLE
Implement Two factor authentication

### ISSUE THREAD

**Issue by fdiluch** [xwiki-labs/cryptpad#329]
Hello,
Could you implement a Implement a Two factor authentication 
kind regards
Francisco

**Comment by agru** [comment]
I totally agree to that. Username and password only is dangerous for a service like Cryptpad. TOTP would be great!

**Comment by secur3gamer** [comment]
Any updates on this? The issue is 2 and a half years old.

**Comment by userliluzibert** [comment]
> Any updates on this? The issue is 2 and a half years old.

Nearly 3 and a half years old for me and still no changes.

**Comment by Mello3vnik** [comment]
> > Any updates on this? The issue is 2 and a half years old.
> 
> Nearly 3 and a half years old for me and still no changes.

This is cryptpad's roadmap : 
https://cryptpad.fr/kanban/#/2/kanban/view/PLM0C3tFWvYhd+EPzXrbT+NxB76Z5DtZhAA5W5hG9wo/
2fa is planned for the end of the year

**Comment by ptman** [comment]
I have very good experience with WebAuthn from another project. Please consider not wasting time on TOTP or email second factor and just go with webauthn. WebAuthn even makes passwordless auth possible. And usernameless (called passkeys or discoverable credentials)

Check out e.g.

https://webauthn.io
https://passkeys.io
https://hanko.io
https://apple-passkey.demo.hanko.io/
